Be more specific about the benefits of P3815

jesswong opened the pull request.

jesswong committed:
Address Ville's feedback

jesswong committed:
Address Ville's feedback

ispeters commented on 3815.md:
```diff
@@ -36,12 +36,13 @@ concept scope_token =
 
-During the development of these sample implementations, it was observed that reintroducing the `scope_association`
-concept from [@P3149R7] would yield several benefits:
+In [@P3149R7], we also introduced the `scope_association` concept as an RAII handle for managing scope associations. However, this facility was removed in [@P3149R9] due to concerns about the unconventional behavior of its copy constructor. This change was made without implementation experience.
```
I'm a _little_ hesitant to say "this change was made without implementation experience"; did we have experience with the association concept? We might have, but I don't remember.

jesswong replied:
At the time of LEWG's approval of R9, I don't think we had an implementation of scopes with a `scope_association` concept. My recollection of when we discovered these issues was when we started working on R10 wording and realized the code looked gross. We then tried implementing the wording with a `scope_association` concept and realized it was way better. 

Maybe I can be more explicit about the fact that _because_ we didn't have implementation experience with `scope_assocation` at the time of R9's approval, the full consequences of its removal were not yet understood.

ispeters commented on 3815.md:
````diff
@@ -109,6 +110,8 @@ struct @_spawn-state_@ : @_spawn-state-base_@ {
 ```
 ::::
 
+In the above example, a key difference when implementing _`spawn`_ with a `scope_association` concept is that `t.try_associate()` is invoked in the constructor rather than in `run()`. This change simplifies exception handling within _`spawn`_: if an exception occurs, only the allocation needs to be explicitly cleaned up. By contrast, if `t.try_associate()` throws from `run()`, both destruction an ...
````
I love this text, but I think it's wrong to italicize `spawn`, right? Or did you mean _`spawn-state`_?
```suggestion
In the above example, a key difference when implementing _`spawn-state`_ with a `scope_association` concept is that `t.try_associate()` is invoked in the constructor rather than in `run()`. This change simplifies exception handling within _`spawn-state`_: if an exception occurs, only the allocation needs to be explicitly cleaned up. By contrast, if `t.try_associate()` throws from `run()`, both destruction and deallocation needs to be explicitly handled.
```

jesswong committed:
Address feedback

jesswong committed:
were -> was for consistency

jesswong committed:
Address nits

ispeters committed:
Changes from LWG session three

Update th wording with LWG feedback

ispeters committed:
Initialize assoc with an IILE

ispeters committed:
Late-night feedback from Tomasz

ispeters committed:
Leave self-assignment unmentioned

ispeters committed:
Apply Tomasz's emailed feedback

Move simple_counting_scope::assoc-t into a class template in
[exec.counting.scopes.general] and update the two scopes' assoc-t nested
types to be nested type aliases to the newly-defined class template.

ispeters committed:
More real-time LWG feedback

ispeters committed:
Final(?) LWG feedback

ispeters committed:
Final changes
